src/Elmish/React/Ref.purs

```diff
@@ -0,0 +1,36 @@
+module Elmish.React.Ref
+  ( Ref
+  , callbackRef
+  )
+  where
+
+import Prelude
+
+import Data.Maybe (Maybe(..))
+import Data.Nullable (Nullable)
+import Data.Nullable as Nullable
+import Effect (Effect)
+import Effect.Uncurried (EffectFn1)
+import Elmish.Dispatch ((<?|))
+import Elmish.Foreign (class CanPassToJavaScript)
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | An opaque type representing the type for React `ref` props
+data Ref (el :: Type)
+
+instance CanPassToJavaScript (Ref a)
+
+-- | Turns a callback function (`el -> Effect Unit`) into a `Ref`. The callback
+-- | function should add the `el` parameter to some state.
+callbackRef :: forall el. Maybe el -> (Maybe el -> Effect Unit) -> Ref el
+callbackRef ref setRef = mkCallbackRef $ setRef <?| \ref' -> case ref, Nullable.toMaybe ref' of
+  Nothing, Nothing -> Nothing
```

Comment:
This surprises me.
I have been using `/\` for tuples, what is this comma syntax?
I was under the impression that the builtin tuple type was just an ordinary data constructor called `Tuple` with no special syntax.

Reply:
The comma is not syntax for tuples. It's syntax for `case`.

Unlike in Haskell, there may be several comma-separated expressions between `case` and `of`, and each alternative must include exactly that many patterns.

Reply:
That’s right — this isn’t a `Tuple`. This is part of the `case … of …` syntax. It supports comma-separated cases since there’s no special `(,)` syntax.

Reply:
@fsoikin beat me to it, again